System.IO.Abstractions/DirectoryInfoBase.cs

```diff
@@ -102,7 +106,7 @@ public abstract class DirectoryInfoBase : FileSystemInfoBase
         {
             if (directoryInfo == null)
                 return null;
-            return new DirectoryInfoWrapper(directoryInfo);
+            return new DirectoryInfoWrapper(new FileSystem(), directoryInfo);
```

Comment:
I can't see much alternative to this given how the implicit cast works.
We could have a readonly static `FileSystem.Instance` field which we encourage people to use, I guess.

(To be honest I'm not a huge fan of implicit casts like this - they make dependency injection harder and tend to hide bugs, as witnessed by the one I fixed as part of this PR in `MockFileInfo`. Part of the public API, though. :-/)